src/Patterns/ProductUpdater.php

```diff
@@ -0,0 +1,361 @@
+<?php
+
+namespace Automattic\WooCommerce\Blocks\Patterns;
+
+/**
+ * Pattern Images class.
+ */
+class ProductUpdater {
+
+	/**
+	 * Generate AI content and assign AI-managed images to Products.
+	 *
+	 * @param array  $vertical_images The vertical images.
+	 * @param string $business_description The business description.
+	 *
+	 * @return bool|\WP_Error True if the content was generated successfully, WP_Error otherwise.
+	 */
+	public function generate_content( $vertical_images, $business_description ) {
+		$last_business_description = get_option( 'last_business_description_with_ai_content_generated' );
+
+		if ( $last_business_description === $business_description ) {
+			return true;
+		}
+
+		$real_products = $this->fetch_product_ids();
+
+		if ( count( $real_products ) > 0 ) {
```

Comment:
Shall we check the data type of `$real_products` since it's possible (albeit unlikely) that [it could return an `object` or `null` as well as an `array` type](https://developer.wordpress.org/reference/classes/wpdb/get_results/#return)?

Reply:
Or maybe just make sure `$this->fetch_product_ids()` _always_ returns an `array`

Reply:
Good call: updated on https://github.com/woocommerce/woocommerce-blocks/commit/3da76eb5ef1ea018cb9f6de71a1b5a2cfabba7cd